lib/mix/lib/mix/tasks/test.ex

```diff
@@ -243,6 +246,13 @@ defmodule Mix.Tasks.Test do
       :noop ->
         :ok
     end
+
+    if opts[:listen_on_stdin] do
+      IO.gets(:stdio, '')
+      Mix.shell.info 'Restarting...'
```

Comment:
Oh jeez, whoops. Good catch. Had just copy/pasted your gist and then got distracted by `getn` not working as I expected.